docs/decisions/0026-file-service.md

```diff
@@ -0,0 +1,148 @@
+---
+# These are optional elements. Feel free to remove any of them.
+status: proposed
+contact: crickman, mabolan, semenshi
+date: 2024-01-16
+---
+
+# File Services
+
+## Context and Problem Statement
+OpenAI provides a file service for uploading files to be used for *assistant retrieval* or *model fine-tuning*: `https://api.openai.com/v1/files`
+
+Other providers may also offer some type of file-service, such as Gemini.
+
+> Note: *Azure Open AI* does not currently support the OpenAI file service API.
+
+## Considered Options
+
+1. Add OpenAI file service support to `Microsoft.SemanticKernel.Experimental.Agents`
+2. Add a file service abstraction and implement support for OpenAI
+3. Add OpenAI file service support without abstraction
+
+## Decision Outcome
+
+> Option 2. **Add a file service abstraction and implement support for OpenAI**
+
+Defining a generalized file service interface provides an extensibility point for other vendors, in addition to *OpenAI*.
+
+## Pros and Cons of the Options
+
+### Option 1. Add OpenAI file service support to `Microsoft.SemanticKernel.Experimental.Agents`
```

Comment:
Is there a middle ground between Option 1 and Option 2 where we provide an abstraction for docs/files (e.g., `DocumentContent`) and leave it up to the AI connector to handle how the file is "managed" (i.e., not introduce `IFileService`)?

Reply:
Thinking hard...ultimately, something needs to call the rest api...I mean, I guess we could mangle `OpenAIChatCompletionService` without changing `IChatCompletionService` or `ITextGenerationService`

Reply:
I think that's Option #3.  I've added signatures for that.

Reply:
Talked to Matthew offline - doing option #3